flake.nix

```
@@ -15,6 +15,21 @@
       # Nixpkgs instantiated for supported system types.
       nixpkgsFor = forAllSystems (system: import nixpkgs { inherit system; });
     in {
+      packages = forAllSystems (system:{
+        # this custom build can be removed once https://github.com/cloudfoundry/bosh-bootloader/issues/596 is implemented.
+        bosh-bootloader = nixpkgsFor.${system}.buildGoModule rec {
+          name = "bosh-bootloader";
+          src = nixpkgsFor.${system}.fetchgit {
+            url = "https://github.com/cloudfoundry/bosh-bootloader";
+            rev = "refs/tags/v9.0.17";
```

The nix-package-name should by convention contain the version. You can do it instead as follows:
```suggestion
          pname = "bosh-bootloader";
          version = "9.0.17";
          src = nixpkgsFor.${system}.fetchgit {
            url = "https://github.com/cloudfoundry/bosh-bootloader";
            rev = "refs/tags/v${version}";
```
Then:
 + `name` will be calculated using `pname` and `version`.
 + Version-part in `name` and downloaded code will always be in sync.
 + It actually makes sense that you use the `rec`-keyword to access within a value of the map another key of the same map.

---

You got me with the `rec`-keyword 💯 😈 , thanks for that!